File: elixir/bob/bob.exs
defmodule Bob do
  def silence?(input) do
    String.strip(input) == ""
  end

  def yelling?(input) do
    Regex.match?(~r/\p{L}/, input) and input == String.upcase(input)
  end

  def question?(input) do
    String.ends_with?(input, "?")
  end

  def hey(input) do
    cond do
      silence?(input)  -> "Fine. Be that way!"
      yelling?(input)  -> "Whoa, chill out!"
      question?(input) -> "Sure."
      true             -> "Whatever."
    end
  end
end

File: nim/bob/bob.nim
import strutils

proc hey*(utterance: string): string =
  if isSpace(utterance) or len(utterance) == 0:
    return "Fine. Be that way!"
  elif contains(utterance, {'A'..'Z'}) and not contains(utterance, {'a'..'z'}):
    return "Whoa, chill out!"
  elif endsWith(utterance, "?"):
    return "Sure."
  else:
    return "Whatever."
